tests/test_harness.rs

```diff
@@ -0,0 +1,75 @@
+use std::{
+    env::{self, join_paths},
+    fs::canonicalize,
+    io::{Error, ErrorKind},
+    process::{Command, ExitStatus},
+};
+
+pub struct Harness;
+
+type Result<T = ()> = std::io::Result<T>;
```

Comment:
I think I would prefer `use std::io::Result` instead. Writing `Result<()>` is not much worse than `Result`.